torch/fx/experimental/symbolic_shapes.py

```diff
@@ -4305,9 +4299,16 @@ def bound_sympy(self, expr: sympy.Expr, size_oblivious: bool = False) -> ValueRa
         var_to_range = {x: self.var_to_range.get(x, None) for x in expr.free_symbols}
         if size_oblivious:
             # Clamp values of size-like variables
+            # NB: discarding the old upper bound in intentional, per
+            # https://github.com/pytorch/pytorch/pull/123675
             for x in self.size_like & var_to_range.keys():
                 if var_to_range[x] is not None:
-                    var_to_range[x] = ValueRanges(2, sys.maxsize - 1)
+                    # TODO: Maybe we should preserve the lower bound here?
+                    # NB: Upper bound the range with some large amount
+                    # (281 TB here was arbitrarily chosen to match addressable
+                    # virtual space on x86_64) so that comparisons against
+                    # sys.maxsize can no-op
+                    var_to_range[x] = ValueRanges(2, 2 ** 48)
```

Comment:
Since we've undone https://github.com/pytorch/pytorch/pull/122372 we need some other way to get around the data dependent error. Size oblivious to the rescue! Assume unbacked is not equal to sys.maxsize.